chain/client/src/stateless_validation/chunk_endorsement_tracker.rs

```diff
@@ -40,33 +55,66 @@ impl ChunkEndorsementTracker {
         Self {
             epoch_manager,
             chunk_endorsements: SyncLruCache::new(NUM_CHUNKS_IN_CHUNK_ENDORSEMENTS_CACHE),
+            // We can use a different cache size if needed, it does not have to be the same as for `chunk_endorsements`.
+            pending_chunk_endorsements: SyncLruCache::new(NUM_CHUNKS_IN_CHUNK_ENDORSEMENTS_CACHE),
         }
     }
 
+    /// Process pending endorsements for the given chunk header.
+    /// It removes these endorsements from the `pending_chunk_endorsements` cache.
+    pub fn process_pending_endorsements(
+        &self,
+        chunk_header: &ShardChunkHeader,
+    ) -> Result<(), Error> {
+        let chunk_hash = &chunk_header.chunk_hash();
+        let chunk_endorsements = {
+            let mut guard = self.pending_chunk_endorsements.lock();
+            guard.pop(chunk_hash)
+        };
+        let chunk_endorsements = match chunk_endorsements {
+            Some(chunk_endorsements) => chunk_endorsements,
+            None => {
+                tracing::debug!(target: "stateless_validation", ?chunk_hash, "No pending chunk endorsements.");
```

Comment:
I think we can remove this trace, I would expect it to be hit with a high probability. Instead we are better off adding a trace for when some chunk endorsements ARE actually present in the pending cache.